Assets/Scripts/Interactables/Dialogue/DialogueInteractable.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DialogueInteractable : Interactable
+{
+    public Conversation conversation;
+
+    private GameObject NPCBubble;
+    private GameObject BirdieBubble;
+    private SpeakerUI speakerUILeft;
+    private SpeakerUI speakerUIRight;
+
+    private int activeLineIndex = 0;
+
+    public override void OnInteract(GameObject birdie)
+    {
+        BirdieBubble = birdie.transform.GetChild(1).gameObject; 
+        NPCBubble = this.transform.GetChild(1).gameObject;
+
+        speakerUILeft = BirdieBubble.GetComponent<SpeakerUI>();
+        speakerUIRight = NPCBubble.GetComponent<SpeakerUI>();

```

Comment:
I think these should be changed to:

```
speakerUILeft = birdie.GetComponentInChildren<SpeakerUI>.gameObject; 
speakerUIRight = GetComponentInChildren<SpeakerUI>.gameObject;
```
If we wanted to program even more defensively we could do `GetComponentsInChildren` above and check that list to make sure there is only 1 component, but that isn't needed 100%.

The hardcoded first child is not gonna end well for us in the long run though haha

Reply:
Yeah definitely not.. thank you!